### PR TITLE
Add CI for openshift/mapo

### PR DESCRIPTION
**Pull request by mandre**
This configure prow to run various tests in the
shiftstack/machine-api-provider-openstack repo:
- unit tests
- gofmt
- govet
- golint
- e2e-openstack

Later, we'll want to move this repo under the openshift organization.

Implements [OSASINFRA-2520](https://issues.redhat.com/browse/OSASINFRA-2520)

### CODE REVIEW

**Comment by mandre** [comment]
@mdbooth PTAL, even though the e2e-openstack job is not passing, it's using [the CI built MAPO image](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/20896/rehearse-20896-pull-ci-shiftstack-machine-api-provider-openstack-master-e2e-openstack/1423008596836225024/artifacts/build-logs/openstack-machine-controllers.log) to deploy as shown in the release payload's [image-references](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/20896/rehearse-20896-pull-ci-shiftstack-machine-api-provider-openstack-master-e2e-openstack/1423008596836225024/artifacts/release/artifacts/release-payload-latest/image-references).
/assign @mdbooth

File: ci-operator/config/shiftstack/machine-api-provider-openstack/shiftstack-machine-api-provider-openstack-master.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    namespace: ocp
+    tag: base
+  ocp_builder_rhel-8-golang-1.16-openshift-4.9:
+    name: builder
+    namespace: ocp
+    tag: rhel-8-golang-1.16-openshift-4.9
+binary_build_commands: make build
+build_root:
+  from_repository: true
+canonical_go_repository: sigs.k8s.io/machine-api-provider-openstack
```
Comment (iamemilio) [inline ci-operator/config/shiftstack/machine-api-provider-openstack/shiftstack-machine-api-provider-openstack-master.yaml:13]: Should this be prepended by `sigs.k8s.io/`? I think its supposed to just be: `shiftstack/machine-api-provider-openstack` as it is right now, but will ideally be `github.com/openshift/machine-api-provider-openstack` once we move it into the openshift org.

**Comment by iamemilio** [comment]
/lgtm
I can see that it manages to get at least one worker node up, and this is clearly working as intended.

**Comment by iamemilio** [comment]
seems like there is a deprovision bug in MAPO: https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/periodic-ci-shiftstack-shiftstack-ci-main-cleanup-vexxhost/1423207361731891200

**Comment by EmilienM** [comment]
can we make it so the job stops to run? this is causing issues with cluster cleanup

/joke

**Comment by iamemilio** [comment]
The error we are seeing for this in the logs does not really make sense. It is throwing an error about being unable to fetch a kubernetes object, but that section of code was never changed... I am going to try to run this locally and see why its getting stuck

**Comment by mandre** [comment]
Give this another try after https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/994 got imported via https://github.com/shiftstack/machine-api-provider-openstack/pull/7.
/test pj-rehearse

**Comment by iamemilio** [comment]
Please dont run until https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1016 merges upstream and is vendored downstream

**Comment by EmilienM** [comment]
/lgtm

The tests run fine now, they fail but they need to be fixed in MAPO.

**Comment by iamemilio** [comment]
we should not have merged this yet, it will leak ports

**Pull request merged**